pkg/controller/sidecarterminator/sidecar_terminator_controller.go

```diff
@@ -136,22 +134,25 @@ func (r *ReconcileSidecarTerminator) doReconcile(pod *corev1.Pod) (reconcile.Res
 	if !isInterestingPod(pod) {
 		return reconcile.Result{}, nil
 	}
-
-	sidecarNeedToExecuteKillContainer, sidecarNeedToExecuteInPlaceUpdate, err := r.groupSidecars(pod)
-
+	vk, err := IsPodRunningOnVirtualKubelet(pod, r.Client)
```

Comment:
is it necessary to check whether node is vk? for non-vk situation, we can also call inplace-update to terminate sidecar if the kruise-daemon is not installed.

Reply:
@furykerry How do we know if there is a kruise-daemon installed or not, it's always just by whether it's a VK node or not.